examples/s3gateway/README.md

```diff
@@ -0,0 +1,152 @@
+
+### Accessing Pachyderm repos from TFJobs and the Tensorflow apis
+
+Once you have deployed both Pachyderm and Kubeflow,
+enabled Pachyderm Enterprise Edition with your activation key,
+and confirmed that you can access the S3 gateway,
+you can easily create a TFJob that can read and write to Pachyderm's version-controlled repositories.
+
+In this example, 
+you will find code that uses the Kubeflow [tfoperator example mnist_with_summaries](https://github.com/kubeflow/tf-operator/tree/master/examples/v1beta2/mnist_with_summaries) images as its base and
+does simple reads and writes to a Pachyderm repo.
+
+The script `tf_job_s3_gateway.py` takes two flags `-b` or `--bucket`, 
+which specifies the `s3://` url to the bucket, and
+`-i` or `--inputpath`, 
+which specifies the local directory or mount point which will be the source of files to copy to the bucket.
+
+In `Dockerfile.tf_job_s3_gateway`, 
+that script is built into a Docker image
+along with the local folder `testdata`, 
+which contains the first chapters of the novels Moby Dick and A Tale of Two Cities.
+```
+$  ls -R testdata/
+testdata/:
+'A Tale of Two Cities'	'Moby Dick'
+
+'testdata/A Tale of Two Cities':
+'Chapter 1 The Period.txt'
+
+'testdata/Moby Dick':
+'Chapter 1 Loomings.txt'
+```
+
+The relevant parts of the code from  `tf_job_s3_gateway.py` are below.
+It simply walks through a local directory,
+copying files to an S3 resource using the Tensorflow `file_io` library.
+It then uses the `file_io.walk()` function to to through the bucket,
+printing out the files to standard output as it goes.
+
+```python
+    s3_path = 's3://' + args.bucket + '/'
+    print("walking {}".format(args.inputpath))
+    for dirpath, dirs, files in os.walk(args.inputpath, topdown=True):   
+      for file in files:
+        newpath = s3_path + dirpath + "/" +  file
+        print("copying {} to {}".format(dirpath + "/" + file, newpath))
+        file_io.copy(dirpath + "/" + file, newpath, True)
+
+    for dirpath, dirs, files in file_io.walk(s3_path +  args.inputpath, True):
+        for file in files:
+            newpath = dirpath + "/" + file
+            print("printing {} in {}  as string: >>{}<<".format(file, dirpath, file_io.read_file_to_string(newpath, False)))
+```
+
+If you want to test this in your own Kubeflow/Pachyderm deployment, 
+you can either build your own container using the `Dockerfile.tf_job_s3_gateway.yml` included with this example or
+use the prebuilt container with this code and sample data hosted in the Pachyderm docker registry.
+You create a TFJob using a TFJob manifest.
+We've included one with this example, 
+`tf_job_s3_gateway.yaml`, 
+which you would use to deploy the TFJob to your Kubeflow app
+after you've edited it for your environment.
+
+Here are the steps you need to take to run this in your environment using our prebuilt images.
```

Comment:
This is a lot of steps with a good amount of nuance. How can we simplify this? 

It seems like it can be broken down into more discreet categories:
1. Add your input data to pachyderm
     - create repo, create branch, add data, whatever.
2. We use the following default namespace, bucket, and repo names. If you have different names for any of these, make the BLAH BLAH changes in the manifest or wherever?
3. deploy the TFJob. You can track progress with...

Reply:
The instructions for deploying Pachyderm with Kubeflow have us in the `pachyderm` namespace, so I want to keep it simple, but still provide information on customization.